lib/board.options.js

```diff
@@ -74,7 +76,15 @@ function Options(arg) {
   } else {
     opts = arg;
 
+    if (opts && opts.pins != null) {
```

Comment:
Something we overlooked: this should also check that `typeof opts.pins === "object"`